### PR TITLE
config: migrate test framework to testify

### PR DESCRIPTION
**Pull request by rleungx**
Signed-off-by: Ryan Leung <rleungx@gmail.com>

### What problem does this PR solve?

Issue Number: Ref #4813.

### What is changed and how does it work?
Just as the title says.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: server/config/config_test.go [hunk 1]
```diff
@@ -20,58 +20,54 @@ import (
...
 
 	"github.com/BurntSushi/toml"
 	. "github.com/pingcap/check"
+	"github.com/stretchr/testify/require"
 	"github.com/tikv/pd/server/storage"
 )
 
 func Test(t *testing.T) {
 	TestingT(t)
 }
```
Comment (JmPotato) [inline server/config/config_test.go:36]: Should be removed.

File: server/config/config_test.go [hunk 2]
```diff
@@ -20,58 +20,54 @@ import (
...
+	re.Nil(tls)
 }
 
-func (s *testConfigSuite) TestBadFormatJoinAddr(c *C) {
+func TestBadFormatJoinAddr(t *testing.T) {
+	re := require.New(t)
 	cfg := NewConfig()
 	cfg.Join = "127.0.0.1:2379" // Wrong join addr without scheme.
-	c.Assert(cfg.Adjust(nil, false), NotNil)
+	re.NotNil(cfg.Adjust(nil, false))
```
Comment (JmPotato) [inline server/config/config_test.go:56]: Better use `re.NoError `here.

File: server/config/config_test.go [hunk 3]
```diff
@@ -20,58 +20,54 @@ import (
...
 	opt, err := newTestScheduleOption()
-	c.Assert(err, IsNil)
+	re.NoError(err)
 	storage := storage.NewStorageWithMemoryBackend()
 	scheduleCfg := opt.GetScheduleConfig()
 	scheduleCfg.MaxSnapshotCount = 10
 	opt.SetMaxReplicas(5)
 	opt.GetPDServerConfig().UseRegionStorage = true
-	c.Assert(opt.Persist(storage), IsNil)
+	re.Nil(opt.Persist(storage))
```
Comment (JmPotato) [inline server/config/config_test.go:70]: Ditto.

File: server/config/config_test.go [hunk 4]
```diff
@@ -80,23 +76,25 @@ func (s *testConfigSuite) TestReloadConfig(c *C) {
 	}()
 
 	newOpt, err := newTestScheduleOption()
-	c.Assert(err, IsNil)
-	c.Assert(newOpt.Reload(storage), IsNil)
+	re.NoError(err)
+	re.Nil(newOpt.Reload(storage))
```
Comment (JmPotato) [inline server/config/config_test.go:80]: Same as above.

File: server/config/config_test.go [hunk 5]
```diff
@@ -446,68 +450,87 @@ wait-store-timeout = "120s"
...
 }
 
-func (s *testConfigSuite) TestConfigClone(c *C) {
+func TestConfigClone(t *testing.T) {
+	re := require.New(t)
+	registerDefaultSchedulers()
 	cfg := &Config{}
 	cfg.Adjust(nil, false)
-	c.Assert(cfg.Clone(), DeepEquals, cfg)
+	re.True(reflect.DeepEqual(cfg.Clone(), cfg))
```
Comment (JmPotato) [inline server/config/config_test.go:502]: `require.Equal` is `reflect.DeepEqual` actually. See https://github.com/tikv/pd/issues/5104.

**Review by JmPotato — APPROVED** [review]

File: server/config/config_test.go [hunk 6]
```diff
@@ -213,10 +212,10 @@ type = "random-merge"
 `
 	cfg = NewConfig()
 	meta, err = toml.Decode(cfgData, &cfg)
-	c.Assert(err, IsNil)
+	re.NoError(err)
 	err = cfg.Adjust(&meta, false)
-	c.Assert(err, IsNil)
-	c.Assert(strings.Contains(cfg.WarningMsgs[0], "Config contains undefined item"), IsTrue)
+	re.NoError(err)
+	re.True(strings.Contains(cfg.WarningMsgs[0], "Config contains undefined item"))
```
Comment (JmPotato) [inline server/config/config_test.go:218]: This could use `re.Contains` to check.

File: server/config/config_test.go [hunk 7]
```diff
@@ -301,35 +302,28 @@ enable-make-up-replica = false
...
+	re.False(cfg.Schedule.EnableRemoveDownReplica)
+	re.False(cfg.Schedule.EnableMakeUpReplica)
+	re.False(cfg.Schedule.EnableRemoveExtraReplica)
 	b, err := json.Marshal(cfg)
-	c.Assert(err, IsNil)
-	c.Assert(strings.Contains(string(b), "disable-replace-offline-replica"), IsFalse)
-	c.Assert(strings.Contains(string(b), "disable-remove-down-replica"), IsFalse)
+	re.NoError(err)
+	re.False(strings.Contains(string(b), "disable-replace-offline-replica"))
+	re.False(strings.Contains(string(b), "disable-remove-down-replica"))
```
Comment (JmPotato) [inline server/config/config_test.go:314]: Ditto.

**Review by JmPotato — CHANGES_REQUESTED** [review]

**Review by JmPotato — APPROVED** [review]

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**